### PR TITLE
Add EC2 SD metrics

### PR DESCRIPTION
**Pull request by dominikschulz**
This PR adds metrics to the AWS EC2 SD, similar to the GCE SD instrumentation.

### CODE REVIEW

File: retrieval/discovery/ec2.go [hunk 1]
```diff
@@ -45,6 +46,33 @@ const (
...
+	ec2SDScrapeFailuresCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "ec2_sd_scape_failures_total",
+			Help:      "The number of EC2-SD scrape failures.",
+		})
+	ec2SDScrapeDuration = prometheus.NewSummary(
+		prometheus.SummaryOpts{
+			Namespace: namespace,
+			Name:      "ec2_sd_scrape_duration",
```
Comment (brian-brazil) [inline retrieval/discovery/ec2.go]: seconds should be in the metric name

File: retrieval/discovery/ec2.go [hunk 2]
```diff
@@ -45,6 +46,33 @@ const (
 	subnetSeparator       = ","
 )
 
+var (
+	ec2SDScrapesCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "ec2_sd_scrapes_total",
```
Comment (brian-brazil) [inline retrieval/discovery/ec2.go]: This is handled by the duration metric

File: retrieval/discovery/ec2.go (hunk 1 above)
Comment (dominikschulz) [inline retrieval/discovery/ec2.go]: Yes, of course. Will do.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**